### ISSUE TITLE
Changing user email address does not trigger any re-verification

### ISSUE THREAD

**Issue by knolleary** [FlowFuse/flowfuse#1058]
### Current Behavior

A user can edit their email via User Settings, but it does not prompt any re-verification.

### Expected Behavior

Changing a user's email address should trigger re-verification.

There are practical questions about what we should do here to help prevent a user locking themselves out. If they typo the email address they have no way to fix the mistake as we then require them to click the link in an email they cannot receive.

We may want to relax what an existing user can do until they reverify - rather than block their access entirely as we do today. For example, add a notification to the page asking them to reverify, but still allow them to continue using the platform.

### Steps To Reproduce

Change your email address.

Observe no attempt is made to reverify and that the Admin view shows the user still has a verified email.

### Environment

- FlowForge version:
- Node.js version:
- npm version:
- Platform/OS:
- Browser:

**Comment by hardillb** [comment]
I have an approach to this, but it results in the user being immediately being directed to the email verification screen when they submit a change of email. It's very abrupt.

Do we think this would be a problem?

**Comment by knolleary** [comment]
> Do we think this would be a problem?

It doesn't fill me with joy.

I sized this thinking primarily about sending the email. But rereading my comments in the issue there is potentially more to do here.

I think we need to handle an unconfirmed email change differently to an unverified email.. which means more state to track.

One possible approach - very much pseudo code so don't take too literally...

 - New model - `UserEmailChange` - `userId`, `old`, `new`, `token`
 - We do not change the User model when then change email - but add an entry to above.
 - The User view gains a new flag (`pending_email_change`) if there is an entry in the `UserEmailChange` table
    - if set, the UI adds a banner to say please verify your new email, rather than force them
    - on user settings page, if flag is set, add a 'resend verification email' button next to email
    - if user changes email again, delete any existing entry in UserEmailChange before adding new one
 - When they verify email, delete the UserEmailChange entry

Happy to pick this up if you want to look at other things.

**Comment by hardillb** [comment]
I'll push it back to todo for now.

**Comment by hardillb** [comment]
This should probably be done in parallel with #1057

**Comment by hardillb** [comment]
I didn't get to this so again looks like it need to get pushed back to 1.2

**Comment by ZJvandeWeg** [comment]
@hardillb Any update on this issue? It's close to the release date, and it's still open.

**Comment by hardillb** [comment]
This is not going to get done this week

**Comment by AllanOricil** [comment]
@hardillb is this still open?

**Comment by hardillb** [comment]
@AllanOricil yes, but as mentioned in the history, the simple fix of just forcing validation straight away feels very harsh. This will probably need some UX design input to come up with what the best solution will be.

**Comment by AllanOricil** [comment]
@hardillb does ff has an UX specialist? Should I wait for him to design the flow, copy other app flow, wait for you to define the flow?
